src/components/v5/common/CompletedAction/partials/PaymentBuilder/partials/PaymentStepDetailsBlock/PaymentStepDetailsBlock.tsx

```diff
@@ -1,57 +1,128 @@
+import { type PaymentStepDetailsBlockProps } from './types.ts';
+import { getSummedTokens } from './utils.ts';
 
-  return allPaid ? (
-    <PaymentOverview
-      className="rounded-lg border border-gray-200 bg-base-white px-[1.125rem] pb-3.5 pt-[1.125rem]"
-      total={[
-        {
-          amount: '500000000000000000000',
-          tokenAddress: '0x90B0A9A9Fe382423aEECBb22e5c1CaEBaeF1Af4C',
-          isClaimed: true,
-        },
-        {
-          amount: '90000000000000000000',
-          tokenAddress: '0x90B0A9A9Fe382423aEECBb22e5c1CaEBaeF1Af4C',
-          isClaimed: true,
-        },
-      ]}
-      paid={[
-        {
-          amount: '500000000000000000000',
-          tokenAddress: '0x90B0A9A9Fe382423aEECBb22e5c1CaEBaeF1Af4C',
-          isClaimed: true,
-        },
-        {
-          amount: '90000000000000000000',
-          tokenAddress: '0x90B0A9A9Fe382423aEECBb22e5c1CaEBaeF1Af4C',
-          isClaimed: true,
-        },
-      ]}
-      payable={[
-        {
-          amount: '500000000000000000000',
-          tokenAddress: '0x90B0A9A9Fe382423aEECBb22e5c1CaEBaeF1Af4C',
-          isClaimed: true,
+const PaymentStepDetailsBlock: FC<PaymentStepDetailsBlockProps> = ({
+  expenditure,
+  refetchExpenditure,
+}) => {
+  const { colony } = useColonyContext();
+  const { currentBlockTime: blockTime, fetchCurrentBlockTime } =
+    useCurrentBlockTime();
+  const { slots = [], finalizedAt, nativeId } = expenditure || {};
+
+  const claimablePayouts = useMemo(
+    () => getClaimableExpenditurePayouts(slots, blockTime, finalizedAt),
+    [blockTime, finalizedAt, slots],
+  );
+
+  const allPayouts = useMemo(
+    () =>
+      slots.flatMap(
+        ({ payouts, claimDelay, id }) =>
+          payouts?.map((payout) => ({
+            ...payout,
+            slotId: id,
+            claimDelay: claimDelay || '0',
+          })) || [],
+      ) || [],
+    [slots],
+  );
+  const groupedData = groupBy(allPayouts, 'tokenAddress');
+  const totals = getSummedTokens(groupedData);
+
+  const paid = sortPayouts(
+    Object.values(groupedData).flatMap((group) =>
+      group.reduce<ExpenditurePayout>(
+        (result, item) => {
+          const { amount, tokenAddress, isClaimed } = item;
+
+          if (!amount || !isClaimed) {
+            return result;
+          }
+
+          return {
+            tokenAddress,
+            amount: BigNumber.from(result.amount || '0')
+              .add(BigNumber.from(amount))
+              .toString(),
+            isClaimed,
+          };
         },
         {
-          amount: '90000000000000000000',
-          tokenAddress: '0x90B0A9A9Fe382423aEECBb22e5c1CaEBaeF1Af4C',
-          isClaimed: true,
+          tokenAddress: '',
+          amount: '0',
+          isClaimed: false,
         },
-      ]}
+      ),
+    ),
+  );
+
+  const summedClaimablePayouts = getSummedTokens(
+    groupBy(claimablePayouts, 'tokenAddress'),
```

Comment:
Could `getSummedTokens` do `groupBy` internally? it would make this component a bit more readable

Reply:
Done